File: JavaScripts/numbers/squareEveryDigit.js
// Square Every Digit

// Create a function that squares every digit of a number.
// Examples
function squareDigits(num){
  // convert number to string
  let x = num.toString()
  // console.log(x);
  // split string
  let y = x.split("");
  // console.log(y)
  // convert string[each] number
  // square number 
  let z = []
  y.forEach(element => z.push(Math.pow(element, 2)))
  // console.log(z)
  return parseInt(z.join(""));
  
  // convert number to string
  // add strings 
  // convert string to num

}
console.log(squareDigits(9119))// ➞ 811181

console.log(squareDigits(2483))// ➞ 416649

console.log(squareDigits(3212))// ➞ 9414

// Notes

// The function receives an integer and must return an integer.

File: ruby/Numbers/square_every_digit.rb
# Square Every Digit
# Create a function that squares every digit of a number.
def square_digits(num)
  str = num.to_s.split('')
  product = ""
  str.each do |c|
    product += (c.to_i * c.to_i).to_s
  end
  return product.to_i
end


# Examples
puts square_digits(9119) #➞ 811181

puts square_digits(2483) #➞ 416649

puts square_digits(3212) #➞ 9414
# Notes
# The function receives an integer and must return an integer.